src/execution_plan/ops/op_merge.c

```diff
@@ -257,25 +264,21 @@ static Record MergeConsume(OpBase *opBase) {
 	if(op->bound_variable_stream) OpBase_PropagateFree(op->bound_variable_stream);
 	OpBase_PropagateFree(op->match_stream);
 
-	/* Since op_merge is is closer to the results op then op_merge_create, it is initially may be marked as the last writer
-	 * Now that it knows that op_merge_create will write after the merge_op write it should try to replace itself as the
-	 * last writer. */
-
-	if(must_create_records)	QueryCtx_UpdateLastWriter((OpBase *)op, op->create_stream);
-
-	// If we are setting properties with ON MATCH, execute all pending updates.
-	_UpdateProperties(op, op->output_records);
 
 	// Exhaust Create stream if we have at least one pattern to create.
 	if(must_create_records) {
 		/* We've populated the Create stream with all the Records it must read;
 		 * pull from it until we've retrieved all newly-created Records. */
+		if(!op->created_records) op->created_records = array_new(Record, 32);
```

Comment:
Wouldn't you prefer creating `created_records` in op's constructor?

Reply:
no
you start with `_handoff`. It will check for the exsitance of `created_records`, and will return null